### PR TITLE
Refactor File and Rank structs

### PR DESCRIPTION
**Pull request by default-username-852**
The ```Rank``` and ```File``` structs now support the ```From``` and ```TryFrom``` traits.

Solves issue #7

### CODE REVIEW

File: src/board.rs [hunk 1]
```diff
@@ -216,8 +217,8 @@ impl Board {
 			let capture_moves = piece.capture_moves();
 
 			for i in 0..capture_moves.len() {
-				let mut curr_rank: isize = square.rank.as_isize();
-				let mut curr_file: isize = square.file.as_isize();
+				let mut curr_rank = isize::from(square.rank);
+				let mut curr_file= isize::from(square.file);
```
Comment (weerox) [inline src/board.rs]: You missed a space there. 

Also, why are you using `::from()` here but `.into()`above? I don't favour one over the other, but they should be the same, for my peace of mind, so choose freely :)

File: src/rank.rs [hunk 2]
```diff
@@ -1,34 +1,32 @@
...
-			7 => '8',
-			_ => ' ',
+impl TryFrom<usize> for Rank {
+	type Error = ();
+
+	fn try_from(value: usize) -> Result<Self, Self::Error> {
+		if value >= 0 && value < 8 {
+			return Ok(Rank(value));
 		}
+		Err(())
```
Comment (weerox) [inline src/rank.rs]: I do like this way of handling branches, rather than my unnecessary else-clause...

File: src/rank.rs [hunk 3]
```diff
@@ -37,21 +35,36 @@ impl Rank {
 			'6' => Ok(Rank(5)),
 			'7' => Ok(Rank(6)),
 			'8' => Ok(Rank(7)),
-			_ => Err(()),
+			_ => Err(())
```
Comment (weerox) [inline src/rank.rs]: I miss the comma, please give it back 😢

**Comment by weerox** [comment]
Also, while I appreciate your `.idea` addition to the `.gitignore`, I'd like to have it in a separate commit, since I intend to squash and merge this PR.

File: src/rank.rs (hunk 3 above)
Comment (default-username-852) [reply src/rank.rs]: Solved in e589e44917d29f611dd92465ca167ca612bad5dc

File: src/rank.rs (hunk 2 above)
Comment (default-username-852) [reply src/rank.rs]: 👍

File: src/board.rs (hunk 1 above)
Comment (default-username-852) [reply src/board.rs]: You can use ```.into()``` when the compiler can infer what type it should convert to from context. When the compiler can't infer types, you have to specify by using the ```::from()``` method.

I changed it to use into in 99b1165ad869c0b444a7166ef44671f67762e0d2 for the sake of consistency.

File: src/board.rs [hunk 4]
```diff
@@ -216,8 +217,8 @@ impl Board {
 			let capture_moves = piece.capture_moves();
 
 			for i in 0..capture_moves.len() {
-				let mut curr_rank: isize = square.rank.as_isize();
-				let mut curr_file: isize = square.file.as_isize();
+				let mut curr_rank: isize = square.rank.into();
+				let mut curr_file: isize= square.file.into();
```
Comment (weerox) [inline src/board.rs]:
```suggestion
				let mut curr_file: isize = square.file.into();
```

**Pull request merged**